collabora_online.module

```diff
@@ -123,24 +123,59 @@ function collabora_online_entity_operation(EntityInterface $entity) {
  * Checks access for the new media operations provided by this module.
  */
 function collabora_online_media_access(MediaInterface $media, string $operation, AccountInterface $account): AccessResultInterface {
-    $type = $media->bundle();
     switch ($operation) {
         case 'preview in collabora':
-            return AccessResult::allowedIfHasPermission($account, "preview $type in collabora");
+            $type = $media->bundle();
+            if ($media->isPublished()) {
+                return AccessResult::allowedIfHasPermission($account, "preview $type in collabora")
+                    ->addCacheableDependency($media);
+            }
+            $preview_own_permission = "preview own unpublished $type in collabora";
+            if (!$account->hasPermission($preview_own_permission)) {
+                return AccessResult::neutral()
+                    ->cachePerPermissions()
+                    ->addCacheableDependency($media)
+                    ->setReason("The user does not have the '$preview_own_permission' permission.");
+            }
+            // Use '==' because Drupal sometimes loads integers as strings.
+            $is_owner = ($account->id() && $account->id() == $media->getOwnerId());
+            if ($is_owner) {
+                $access_result = AccessResult::allowed();
+            }
+            else {
+                $access_result = AccessResult::neutral()
+                    ->setReason("The user has the '$preview_own_permission' permission, but is not the owner of the media item.");
+            }
+            return $access_result
+                ->cachePerPermissions()
+                ->cachePerUser()
```

Comment:
Why  user and permissions? user is more granular.

Reply:
I think you are right.
If we add permissions to a role, does the access cache get reset for all users with that role?
Same if we add a role to a user?

Reply:
I just checked in core, in most places with `->cachePerUser()` there is no `->cachePerPermission()`.
So this seems the safe thing to do.